### DISCUSSION TITLE
Support React 18 in v7

### DISCUSSION THREAD

**Discussion by artola** [gpbl/react-day-picker#2448] [Ideas]
We are using v7 where the peerDependency are below 18.

https://github.com/gpbl/react-day-picker/blob/bf66a18fbcef3b5438b182a39157ba996da0a4e8/package.json#L58

Could be possible to add `|| 18.0.0`? We know that it works, but we have almost 50 warnings in the console.

```text
react is listed by your project with version 18.2.0 (p7513b), which doesn't satisfy what react-day-picker (via @sl/fusion.common-components) and other dependencies request (but they have non-overlapping ranges!).
```

**Comment by gpbl** [comment]
@artola v7 is frozen so there are no plans to update it. You can however override your dependencies using your package manager:

- For `npm`: see https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
- For `pnpm`: https://pnpm.io/package_json#pnpmoverrides
- For `yarn`: see https://classic.yarnpkg.com/en/docs/selective-version-resolutions/

Let us know if this works for you!

**Reply by artola** [reply]
> @artola v7 is frozen so there are no plans to update it. You can however override your dependencies using your package manager:
> 
> * For `npm`: see https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
> * For `pnpm`: https://pnpm.io/package_json#pnpmoverrides
> * For `yarn`: see https://classic.yarnpkg.com/en/docs/selective-version-resolutions/
> 
> Let us know if this works for you!

I use Yarn and it is not possible to "override" `peerDependencies`. If a value is set, then it is not applied any override. Because of this we have 50+ warnings, that we can just hide using `logFilters` but still will print:

```
➤ YN0000: ┌ Post-resolution validation
➤ YN0086: │ Some peer dependencies are incorrectly met by dependencies; run yarn explain peer-requirements for details.
➤ YN0000: └ Completed
```

I know that this lib is on v9 already, but v8 and v9 introduce too many breaking changes and therefore it is not a trivial change the version number upgrade.
